src/lightning_app/core/app.py

```diff
@@ -280,9 +280,13 @@ def _collect_deltas_from_ui_and_work_queues(self) -> t.List[t.Union[Delta, APIRe
                 component_output: t.Optional[ComponentDelta] = self.get_state_changed_from_queue(self.delta_queue)
                 if component_output:
                     logger.debug(f"Received from {component_output.id} : {component_output.delta.to_dict()}")
-                    work = self.get_component_by_name(component_output.id)
-                    new_work_delta = _delta_to_app_state_delta(self.root, work, deepcopy(component_output.delta))
-                    deltas.append(new_work_delta)
+                    try:
+                        work = self.get_component_by_name(component_output.id)
+                        new_work_delta = _delta_to_app_state_delta(self.root, work, deepcopy(component_output.delta))
+                        deltas.append(new_work_delta)
+                    except (KeyError, AttributeError):
+                        # The component has been deleted.
+                        pass
```

Comment:
With this exception handling, we don't know whether the errors came from the first or the second line. The skipping here will make it almost impossible to debug any bugs here, because a skipped error here means another error will surface somewhere else, and thus can be misleading from the real source problem. 

If it's possible, I suggest to check here that `component_output.id` exists, so the exception handling here can be avoided?

Reply:
Yes, makes sense.